test/SharedFx.UnitTests/SharedFxTests.cs

```diff
@@ -36,40 +37,45 @@ public async Task BaselineTest(SharedFxConfig config)
 
-            if (!Directory.Exists(AppContext.BaseDirectory + "unzipped"))
+            try
             {
-                ZipFile.ExtractToDirectory(AppContext.BaseDirectory, "unzipped");
-            }
-
-            var nugetAssembliesPath = Path.Combine(AppContext.BaseDirectory, "unzipped", "shared", config.Name, previousVersion);
+                Directory.CreateDirectory(tempDirectoryPath);
+                ZipFile.ExtractToDirectory(zipPath, tempDirectoryPath);
+                var nugetAssembliesPath = Path.Combine(tempDirectoryPath, "shared", config.Name, previousVersion);
 
-            var files = Directory.GetFiles(nugetAssembliesPath, "*.dll");
-            foreach (var file in files)
-            {
-                try
+                var files = Directory.GetFiles(nugetAssembliesPath, "*.dll");
+                foreach (var file in files)
                 {
-                    var assemblyVersion = AssemblyName.GetAssemblyName(file).Version;
-                    var dllName = Path.GetFileName(file);
-                    nugetAssemblyVersions.Add(dllName, assemblyVersion);
+                    try
+                    {
+                        var assemblyVersion = AssemblyName.GetAssemblyName(file).Version;
+                        var dllName = Path.GetFileName(file);
+                        nugetAssemblyVersions.Add(dllName, assemblyVersion);
+                    }
+                    catch (BadImageFormatException) { }
                 }
-                catch (BadImageFormatException) { }
-            }
 
-            files = Directory.GetFiles(dir, "*.dll");
+                files = Directory.GetFiles(dir, "*.dll");
 
-            Assert.All(files, file =>
-            {
-                try
+                Assert.All(files, file =>
                 {
-                    var localAssemblyVersion = AssemblyName.GetAssemblyName(file).Version;
-                    var dllName = Path.GetFileName(file);
-                    Assert.True(nugetAssemblyVersions.ContainsKey(dllName), $"Expected {dllName} to be in the downloaded dlls");
-                    Assert.True(localAssemblyVersion.CompareTo(nugetAssemblyVersions[dllName]) >= 0, $"Expected the local version of {dllName} to be greater than or equal to the already released version.");
-                }
-                catch (BadImageFormatException) { }
+                    try
+                    {
+                        var localAssemblyVersion = AssemblyName.GetAssemblyName(file).Version;
+                        var dllName = Path.GetFileName(file);
+                        Assert.True(nugetAssemblyVersions.ContainsKey(dllName), $"Expected {dllName} to be in the downloaded dlls");
+                        Assert.True(localAssemblyVersion.CompareTo(nugetAssemblyVersions[dllName]) >= 0, $"Expected the local version of {dllName} to be greater than or equal to the already released version.");
```

Comment:
In a similar vein, shouldn't this check be `Assert.InRange(localAssemblyVersion.CompareTo(nugetAssemblyVersions[dllName]), 0, int.MaxValue)`?

Reply:
I think the `Assert.Contains` change is fine but I'd like to at least keep this Assert.True to keep the error message.

Reply:
I find custom `Assert.X(...)` error messages don't help much. But, I defer to @Eilon here&hellip;

Reply:
Okay, I updated it to use `Assert.InRange`. Merging now.